WordPress/gutenberg
# Fix for bug with toggling opensInNewWindow state

## Description
Currently the opensInNewWindow state is not toggled in InlineLinkUI if both the activeAttributes.url and activeAttributes.target props both change - in this case only the url change is reflected in the component state due to getDerivedStateFromProps returning before the opensInNewWindow check is hit

## How has this been tested?

Tested manually as follows:

- Apply this patch to your local gutenberg dev env
- From a post/page editor, create a paragraph block
- Add a link
- Set the link to open in a new tab
- Click anywhere else
- Click the link -> settings , the “open in a new tab” should still be toggled on
- Save the page/post and reload the editor
- Click the link -> settings , the “open in a new tab” should still be toggled on
- Repeat the above steps with a toggled off state and ensure it also persists correctly

Also added a unit test to check that state.opensInNewWindow is set when props.activeAttributes.target _blank passed in.

## Types of changes
Fixes #17770

## Checklist:
- [ x ] My code is tested.
- [  x ] My code follows the WordPress code style. 
- [  NA ] My code follows the accessibility standards.
